src/seckrit.py

```diff
@@ -58,15 +73,31 @@ def create_parent_dirs(path):
     os.makedirs(os.path.dirname(path), exist_ok=True)
 
 
-def load_manifest(path):
+def load_manifest_from_file(path):
     """
-    Loads and validates the YAML manifest at the given path,
-    returning a dictionary containing instructions for returning secrets.
+    Loads the YAML manifest at the given path,
+    sends the YAML for validation and returns the result.
     """
     # Load the manifest file.
     with open(path) as manifest_file:
         manifest = yaml.safe_load(manifest_file)
+    return validate_manifest(manifest)
+
 
+def load_manifest_from_string():
+    """
+    Loads the YAML manifest from the environment variable,
+    sends the YAML for validation and returns the result.
+    """
+    manifest = yaml.safe_load(os.environ.get('SECKRIT_MANIFEST'))
+    return validate_manifest(manifest)
+
+
+def validate_manifest(manifest):
+    """
+    Validates the YAML manifest being passed as argument,
```

Comment:
```suggestion
    Validates the given manifest dictionary,
```
The manifest isn't really YAML anymore once loaded (it's a Python `dict`).